pyproject.toml

```
@@ -72,7 +72,7 @@ SQLAlchemy-Utils = "!=0.36.8"  # https://github.com/kvesteri/sqlalchemy-utils/is
 sqlitedict = "*"
 sqlparse = "*"
 starlette = "*"
-starlette-context = "*"
+starlette-context = ">=0.3.1"  # prevent downgrading by Poetry's dependency resolver
```

I think that poetry chooses version 0.1.3 for starlette-context because it is the latest version that didn't enforce `python_requires=">=3.7"` (see https://github.com/tomwojcik/starlette-context/commit/79ca60c1fd4bff5aec3f0a66e9a43bb053ee5812#diff-60f61ab7a8d1910d86d9fda2261620314edcae5894d5aaa236b821c7256badd7R14 ), although all its releases have 3.7 as the minimum Python version in the classifier tags.

```suggestion
starlette-context = {version = "*", python = ">=3.7"}
```

---

Hmm.. I'll try running the update locally right now to verify this works. I must admit I didn't look that deep into the cause of Poetry's choice. Now I will..

---

Verified - this works!